lnrpc/rpc.proto

```diff
@@ -1774,8 +1783,23 @@ message OpenChannelRequest {
 }
 message OpenStatusUpdate {
     oneof update {
+        /**
+        Signals that the channel is now fully negotiated and the funding
+        transaction published.
+        */
         PendingUpdate chan_pending = 1;
+
+        /**
+        Signals that the channel's funding transaction has now reached the
+        required number of confirmations on chain and can be used.
+        */
         ChannelOpenUpdate chan_open = 3;
+
+        /**
```

Comment:
Relying on the streaming response may make this harder to use over REST. Have you considered instead making the workflow primarily drive a state machine via `FundingStateStep`? Either way though, it seems like we need a new response here.

Reply:
I think this can be done by adding a new RPC `OpenChannelPSBTSync` that works very similar to `OpenChannelSync` but already returns once the `OpenStatusUpdate_PsbtFund` message arrives.
The rest can then be done through the transition messages. I'll add a status type to the funding intent to turn it into a "real" state machine and add a new message like `FundingTransitionMsg_Status` to query it.
I should have everything ready by tomorrow.

Reply:
Haven't reviewed the latest version yet, but why couldn't it just use `OpenChannelSync` as is? In any case, REST is secondary, and this is a newer experimental API, so I don't think it's a deal breaker if the initial version doesn't support REST.